site/en/docs/extensions/mv3/xhr/index.md

```diff
@@ -86,56 +79,31 @@ When using resources retrieved via `XMLHttpRequest`, your background page should
 fall victim to [cross-site scripting][9]. Specifically, avoid using dangerous APIs such as the
 below:
 
-```js
-var xhr = new XMLHttpRequest();
-xhr.open("GET", "https://api.example.com/data.json", true);
-xhr.onreadystatechange = function() {
-  if (xhr.readyState == 4) {
-    // WARNING! Might be evaluating an evil script!
-    var resp = eval("(" + xhr.responseText + ")");
-    ...
-  }
-}
-xhr.send();
-```
 
 ```js
-var xhr = new XMLHttpRequest();
-xhr.open("GET", "https://api.example.com/data.json", true);
-xhr.onreadystatechange = function() {
-  if (xhr.readyState == 4) {
+const response = await fetch("https://api.example.com/data.json");
+const jsonData = await response.json();
     // WARNING! Might be injecting a malicious script!
-    document.getElementById("resp").innerHTML = xhr.responseText;
+    document.getElementById("resp").innerHTML = jsonData;
```

Comment:
@IanStanion-google out of curiosity - shouldn't this count as executing remote hosted code? Does this even work in MV3?

Reply:
It seems like this falls under the guidance to [externalize logic with a remove service](https://developer.chrome.com/docs/extensions/migrating/improve-security/#remove-remote-code:~:text=Externalized%20logic%20with%20a%20remote%20service). If there's a subtlety I'm missing, then somewhere on our site, we need to provide clarification.